### ISSUE TITLE
To execute Example for sentiment in GPU and CPU (apple silicon M1)

### ISSUE THREAD

**Issue by ethiel** [deeplearning4j/deeplearning4j#9916]
#### Issue Description
As an apple silicon user, I would like to execute code in the GPU and CPU to take advantage of the Apple's ARM architecture. I tried to run the Word2VecSentimentRNN example but it took hours before finishing. I saw the monitor to see what was happening and I saw the code is executed in the CPU, only.

Please describe our issue, along with:
- expected behavior
The code should be executed in the GPU and CPU
- encountered behavior
Code is only executed in CPU, which is painfully slow. 

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version
1.0.0-M2.1
* Platform information (OS, etc)
Apple silicon, M1

**Comment by treo** [comment]
The Apple Silicon GPU + NPU aren't currently supported.

**Comment by ethiel** [comment]
Hi, @treo thanks for the quick response. I guess the issue can be closed.

**Comment by agibsonccc** [comment]
@ethiel  I'll keep this open as a feature request. I'd have to look in to it. We do have armcompute bindings that may use that but I don't know if those even run on mac or how arm's software would run on something not specifically apple silicon.

**Comment by ethiel** [comment]
@agibsonccc I may be able to help if you explain me what you want to do and how.

**Comment by agibsonccc** [comment]
@ethiel  that unfortunately would require potentially writing some integrations or switching some c++ code around to also run on mac. This would require looking at: https://github.com/deeplearning4j/deeplearning4j/tree/3f93acc1ec943153e00538415aa4704866cc2e00/libnd4j/include/ops/declarable/platform/armcompute
|

Since the ops implemented here (basically we switch out generic kernels for arm compute optimized ones) are mainly for arm based embedded vision systems it might not matter much for mac.
There's also a matter of looking out there for if we need to implement a potentially new integration here. If it's not armcompute it might be a new  (maybe vulkan based?) library that knows how to work with the mac's npu. That appears to be this? https://developer.apple.com/documentation/mlcompute

**Comment by ethiel** [comment]
Ok. I guess you can close it. I'll take a look and I'll ask for reopening this ticket if I come with a solution to the issue. I'm afraid it will take some time. Thanks, @agibsonccc, as usual

**Comment by agibsonccc** [comment]
@ethiel  it's up to you :) I'll just leave this open as a feature request so people can find it and comment on it if they want it. IF you want to take some time to contribute I can walk you through it. There's some unanswered questions how we would offload to the gpu. If we can figure that out I do believe it should be doable. What's nice about the way the c++ library is setup is it can fall back to a default kernel for anything not covered by the gpu routines here.

**Comment by ethiel** [comment]
@agibsonccc let's leave it open. And, please, give me the guidelines to contribute. If I have the time, I'll do it.

**Comment by muthenberg** [comment]
Any new Information on this yet?

**Comment by agibsonccc** [comment]
@muthenberg not really.  M1 execution works but most of the commercial activity around the framework in production cares mainly about linux/cuda and windows for testing. Unless there's a sponsored stakeholder or overwhelming demand for this it's not likely to get done for a while.

**Comment by agibsonccc** [comment]
Closing — this issue references architecture (old UI, M1/M2.1 release, Spark, TF import, ParallelWrapper training, or legacy subsystem) that has been fundamentally rewritten in the upcoming release. If you still experience a similar issue with the new version, please open a new issue with reproduction steps.
